### DISCUSSION TITLE
Tasmota ADS1115 dump load controller

### DISCUSSION THREAD

**Discussion by dexterbot80** [arendst/Tasmota#22027] [Ideas]
Hello teachers!
Could we brake a wind turbine using tasmota and several dump resistors to keep the output voltage of the generator constant for example 60V?

Let's say that the wind increases in intensity and the voltage increases to 61V - can we set a rule to activate digital pin 2 for example?
If the voltage continues to increase, can we also activate digital pin 3, 4, 5?
Would it be more appropriate to control a mosfet bridge to feed a dump resistor to keep the generator voltage constant by braking using the PWM signal?

Thank you

**Comment by sfromis** [comment]
Without trying to comment on soundness of the electrical side, or what might be a better solution, it would certainly be possible to have Tasmota control on/off state of 4 gpio pins, depending on a value from an ADS1115. One question might still be expectations of fast reaction time.

While using rules to control it should be feasible, I'd suggest using the Berry language available with Tasmota on ESP32, as it provides better options for programming details of how it is to work, including how often values are sampled, still depending on the driver.

**Comment by dexterbot80** [comment]
Yes, probably using the PWM output a dumpload with a very smooth action on the generator could be achieved.
It would be ideal if you could help me in this regard - I'm sure there are other interested users.
I think that if we used a voltage divider we could also control higher voltage generators....

If we have this dump load controller maybe is possible to connect an high voltage wind turbine generator to an solar inverter MPPT 500V or more :)

So in the code we must be able to set the maximum voltage that the generator can provide without damaging the MPPT used.
And if, for example, we have a 1000W wind generator, we can use a 2000W dump resistor to brake gradually depending on the output voltage.
